cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -38,27 +41,24 @@
 /**
  *  Handles creating indexes for a given datastore.
  */
-class IndexCreator {
+public class IndexCreator {
 
-    private final SQLDatabase database;
     private final Datastore datastore;
     private static Random indexNameRandom = new Random();
 
-    private final ExecutorService queue;
+    private final SQLDatabaseQueue queue;
 
     private static final Logger logger = Logger.getLogger(IndexCreator.class.getName());
 
-    public IndexCreator(SQLDatabase database, Datastore datastore, ExecutorService queue) {
+    public IndexCreator(Datastore datastore, SQLDatabaseQueue queue) {
```

Comment:
I don't think that is the best approach, it only needs the queue to perform the operations, it doesn't require any of the code from Datastore, so its introducing coupling that isn't required.